everytime/everytime/src/main/java/com/ceos19/everytime/dto/CommentDTO.java

```diff
@@ -0,0 +1,41 @@
+package com.ceos19.everytime.dto;
+
+import com.ceos19.everytime.domain.AboutPost.Comment;
+import com.ceos19.everytime.domain.AboutUser.User;
+import jakarta.persistence.Column;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+public class CommentDTO {
+
+    private String contents;
+
+    private Long likeNum;
+
+    private boolean isDeleted;
+
+    private boolean isReported;
+
+    private Comment parentComment;
+
+    private User user;
+
+    public Comment toComment() {  //생성자를 통해 객체 생성
+        return Comment.builder()
+                .contents(contents)
+                .likeNum(likeNum)
+                .isDeleted(isDeleted)
+                .isReported(isReported)
+                .parentComment(parentComment)
+                .user(user)
+                .build();
+    }
+}
```

Comment:
자바 14에 등장한 record를 사용하면 다음과 같이 간결하게 구현할 수 있어요!

```suggestion
@Builder
public record CommentDTO (String contents, 
                          Long likeNum, 
                          boolean isDeleted, 
                          boolean isReported, 
                          Comment ParentComment, 
                          User user) {
    public Comment toComment() {
        return Comment.builder()
                .contents(contents)
                .likeNum(likeNum)
                .isDeleted(isDeleted)
                .isReported(isReported)
                .parentComment(parentComment)
                .user(user)
                .build();
    }
}
```

record가 불변성과 표준 메서드(equals and hashcode, toString)을 자동으로 제공하기 때문에 저는 주로 dto를 record로 사용하고 있는데, record를 사용해보는 것은 어떤가요?

Reply:
저도 알아갑니다.!!... 이런 것도 있었군요!! 메모 해갑니다.

Reply:
오 이런 🍯 꿀 정보 정말 감사합니다 :) 앞으로 유용하게 쓰도록 하겠습니당 😃